src/sdg/producten/models/product.py

```diff
@@ -79,6 +84,52 @@ def upn_uri(self):
     def upn_label(self):
         return self.upn.upn_label
 
+    @property
+    def upn_is_verweiderd(self):
+        return self.upn.is_verwijderd
+
+    @property
+    def is_sdg_product(self):
+        return bool(self.upn.sdg)
+
+    @cached_property
+    def referentie_product_publicatie_datum(self):
+        try:
+            return bool(
+                Product.objects.get(
+                    catalogus__is_referentie_catalogus=True, generiek_product=self
+                ).active_version
+            )
+        except Product.DoesNotExist:
+            return ProductStatus.labels.MISSING
```

Comment:
This is too magic. Better to return `None` and convert this to the label you need in the status.

```suggestion
            return None
```